runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/DoFnOperator.java

```diff
@@ -739,9 +741,12 @@ public final void processWatermark1(Watermark mark) throws Exception {
     }
 
     currentInputWatermark = mark.getTimestamp();
+    processInputWatermark(true);
+  }
 
+  private void processInputWatermark(boolean advanceInputWatermark) throws Exception {
```

Comment:
Please remove if we don't want to scatter the logic about watermark advancement.

Reply:
I think that is what we must do. We could call `processWatermark` from `processElement`, but the call does stuff not necessary to do in `processElement`. That's why I simply wrapped what is needed to be done in `processInputWatermark` (maybe we could find a better name to make it clearer).